src/contextProviders/Web3Provider.jsx

```diff
@@ -293,12 +107,13 @@ class Web3Provider extends Component {
       validProvider,
       isHomeNetwork,
       isForeignNetwork,
-      isEnabled,
-      setupTimeout,
       showForeignNetRequiredWarning,
       showHomeNetRequiredWarning,
+      web3,
     } = this.state;
 
+    const isEnabled = !!account && !!balance;
```

Comment:
Is `&& !!balance` needed here?

Reply:
We have four subscriptions: web3, network, address and balance. First web3 become ready and last balance will be ready. On each of these four, App will be rerendered by Web3 Context Provider. I think `!!balance` is enough but to make sure we can include all four subscriptions in that conditions.